src/hotspot/cpu/x86/gc/z/zBarrierSetAssembler_x86.cpp

```diff
@@ -468,6 +502,15 @@ class ZSaveLiveRegisters {
     caller_saved.Insert(OptoReg::as_OptoReg(r9->as_VMReg()));
     caller_saved.Insert(OptoReg::as_OptoReg(r10->as_VMReg()));
     caller_saved.Insert(OptoReg::as_OptoReg(r11->as_VMReg()));
+    if (UseAVX > 2) {
```

Comment:
Similar question here: `ZSaveLiveRegisters` handles GP registers and vector registers differently. Why predicate registers are handled the same way GP registers are and not as vector registers?

Reply:
Hi Vladimir, 
ZGC load barriers embeds a CALL instruction which is not exposed to compiler, thus opmask registers which are caller saved as per X86 ABI are saved before making the call and later restored back.

Reply:
I took a closer look at the relevant code and now I better understand what confuses me here.

Through vector registers are caller saved, they aren't added into caller_saved, but just unconditionally spilled. 
Why can't you do the same for KRegisters? `caller_saved.Member(opto_reg)` should always be true for KRegister, shouldn't it?